### PR TITLE
fixes major code smells in GenericUtils.java

### PR DESCRIPTION
**Pull request by hks1**
1. Adds a private constructor
2. Adds missing 'Deprecated' annotation

### CODE REVIEW

File: src/main/java/org/sasanlabs/internal/utility/JSONSerializationUtils.java [hunk 1]
```diff
@@ -17,6 +17,8 @@ public class JSONSerializationUtils {
 
     private static final ObjectMapper MAPPER = new ObjectMapper();
 
+    private static JSONSerializationUtils() {}
```
Comment (preetkaran20) [inline src/main/java/org/sasanlabs/internal/utility/JSONSerializationUtils.java:20]: static constructor?

File: src/main/java/org/sasanlabs/internal/utility/GenericUtils.java [hunk 2]
```diff
@@ -9,11 +9,14 @@ public class GenericUtils {
 
     public static final String LOCALHOST = "127.0.0.1";
 
+    private GenericUtils() {}
```
Comment (preetkaran20) [inline src/main/java/org/sasanlabs/internal/utility/GenericUtils.java:12]: can you please mark this class as final as well?

File: src/main/java/org/sasanlabs/internal/utility/JSONSerializationUtils.java (hunk 1 above)
Comment (preetkaran20) [inline src/main/java/org/sasanlabs/internal/utility/JSONSerializationUtils.java:20]: please mark this class as final.

File: src/main/java/org/sasanlabs/internal/utility/GenericUtils.java (hunk 2 above)
Comment (hks1) [reply src/main/java/org/sasanlabs/internal/utility/GenericUtils.java:12]: sure

File: src/main/java/org/sasanlabs/internal/utility/JSONSerializationUtils.java (hunk 1 above)
Comment (hks1) [reply src/main/java/org/sasanlabs/internal/utility/JSONSerializationUtils.java]: sure

Comment (hks1) [reply src/main/java/org/sasanlabs/internal/utility/JSONSerializationUtils.java]: my bad, that's an oversight. I'll address it in next commit

**Comment by preetkaran20** [comment]
@hks1 Thank you !!!

**Pull request merged**